c++.html.markdown

```diff
@@ -801,6 +801,106 @@ void doSomethingWithAFile(const std::string& filename)
 //   all automatically destroy their contents when they fall out of scope.
 // - Mutexes using lock_guard and unique_lock
 
+///////////////////////////////////////
+// Lambda Expressions (C++11 and above)
+///////////////////////////////////////
+
+// lambdas are a convenient way of defining an anonymous function
+// object right at the location where it is invoked or passed as 
+// an argument to a function.
+
+// Example consider sorting a vector of pairs using the second 
+// value of the pair
+
+vector<pair<int, int> > tester;
+tester.push_back(make_pair(3, 6));
+tester.push_back(make_pair(1, 9));
+tester.push_back(make_pair(5, 0));
+
+// Pass a lambda expression as third argument to the sort function
+// sort is from the <algorithm> header
+
+sort(tester.begin(), tester.end(), [](const pair<int, int> &lhs, const pair<int, int> &rhs) {
+		return lhs.second < rhs.second;
+	});
+
+// Notice the syntax of the lambda expression,
+// [] in the lambda is used to "capture" variables.
+// For Example:
+
+vector<int> dog_ids;
+// number_of_dogs = 3;
+for(int i = 0; i < 3; i++){
+	dog_ids.push_back(i);  
+}
+
+int weight[3];
+weight[0] = 30, weight[1] = 50, weight[2] = 10;
+
+// Say you want to sort dog_ids according to the dogs' weights
+// So dog_ids should in the end become: [2, 0, 1]
+
+// Here's where lambda expressions come in handy
+
+sort(dog_ids.begin(), dog_ids.end(), [weight](const int &lhs, const int &rhs) {
```

Comment:
Better to capture `weight` here by reference. In fact, could you expand on the capture list? Just quick one-line explanations on the syntax for capture by value, by reference, `[=]`, and `[&]`